lucene/core/src/java/org/apache/lucene/codecs/blocktree/BlockTreeTermsReader.java

```diff
@@ -156,6 +190,8 @@ public BlockTreeTermsReader(PostingsReaderBase postingsReader, SegmentReadState
       seekDir(termsIn);
       seekDir(indexIn);
 
+      FSTLoadMode fstLoadMode = FSTLoadMode.valueOf(state.readerAttributes.getOrDefault(FST_MODE_KEY, FSTLoadMode.AUTO.toString()));
```

Comment:
Do we get a nice exception if the string is invalid here?  Can we fix the exception to state which key we were looking up when we got the invalid `FSTLoadMode` enum?

Reply:
I added one.